doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -622,7 +623,180 @@ module sonic-port {
+
+Example:
+```
+container RADIUS_SERVER_STATS {
+    sonic-ext:db-name "COUNTERS_DB";
+    sonic-ext:key-delim ":";
+    config false;
+    list RADIUS_SERVER_STATS_LIST {
+    ....
+    ....
+    }
+}
+```
+
+#### 3. key-pattern: Use this extension when table key follows special pattern.
+
+Example:
+##### ABNF:
+```
+    "QUEUE": {
+        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet120,Ethernet124|0": {
+            "scheduler": "[SCHEDULER|scheduler.1]"
+        },
+        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet100,Ethernet104,Ethernet124|3-4": {
+            "wred_profile": "[WRED_PROFILE|AZURE_LOSSLESS]",
+            "scheduler": "[SCHEDULER|scheduler.0]"
+        }
+    }
+```
+##### YANG:
+```
+container QUEUE {
+    list QUEUE_LIST {
+        key "ifname qindex";
+        sonic-ext:key-pattern "QUEUE|({ifname},)*|{qindex}";
+        
+        leaf ifname {
+            type leafref {
+                path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+            }
+        }
+        leaf qindex {
+            type string;
+        }
+        ....
+    }
+}
+```
+
+#### 4. map-leaf: Mapping tables in Redis are defined using nested 'list'. The outer 'list' is used for multiple instances of mapping. The inner 'list' is used for mapping entries for each outer list instance. Use 'map-leaf' extension to indicate nested 'list' and specify mapping between two fields which are also key fields of inner list.
+
+Example:
+##### ABNF:
+```
+    "DSCP_TO_TC_MAP": {
+        "AZURE_1": {
+            "1": "1",
+            "0": "0",
+            "3": "3",
+            "4": "4"
+        },
+        "AZURE_2": {
+            "50": "1",
+            "51": "0",
+            "52": "3",
+            "53": "4"
+        }
+    }
+```
+##### YANG:
+```
+        container DSCP_TO_TC_MAP {
+            list DSCP_TO_TC_MAP_LIST {
+                key "name";
+                sonic-ext:map-leaf "dscp tc_num"; //every key:value pair is mapped to list keys, e.g. "1":"7" ==> tc_num=1, dscp=7
+
+                leaf name {
+                    type string;
+                }
+
+                list DSCP_TO_TC_MAP { //this is list inside list for storing mapping between two fields
+                    key "dscp tc_num";
+
+                    leaf tc_num {
+                        type string {
+                            pattern "[0-9]?";
+                        }
+                    }
+                    leaf dscp {
+                        type string {
+                            pattern "[1-9][0-9]?|[0-9]?";
+                        }
+                    }
+                }
+            }
+        }
+```
+
+#### 5. custom-validation: Use this extension when customized code is required for validation. Function name has to be given with this extension which will be called to perform custom validation.
```

Comment:
I think you should state that this is specific to CVL, and that the provided function must be provided in the Mgmt Framework common containers. You could also point to the Mgmt Framework HLD for more information on custom validation.